source/fab/steps/archive_objects.py

```diff
@@ -121,12 +121,12 @@ def archive_objects(config: BuildConfig,
 
         if root:
             # we're building an object archive for an executable
-            output_fpath = str(config.build_output / f'{root}.a')
+            output_fpath = Path(str(config.build_output / f'{root}.a'))
```

Comment:
You don't need `Path(str(...))` - `build_output` should be a Path, so using this with `/` and a string gives a path.

Reply:
No, you are right that this should become a Path, not a `str`. Just check:
```
output_fpath = config.build_output / f'{root}.a'
reveal_type(output_fpath)
```
For me this is a `Path`.

Reply:
You missed this line, the `Path(str(...))` is not necessary, `config.build_output / f'{root}.a'` should be a Path already.